src/formattedcode/output_spdx.py

```diff
@@ -312,63 +272,63 @@ def write_spdx(
                         spdx_id = f'LicenseRef-scancode-{license_key}'
                     is_license_ref = spdx_id.lower().startswith('licenseref-')
 
-                    if not is_license_ref:
-                        spdx_license = License.from_identifier(spdx_id)
-                    else:
-                        spdx_license = ExtractedLicense(spdx_id)
-                        spdx_license.name = file_license.short_name
+                    spdx_license = licensing.parse(spdx_id)
+
+                    if is_license_ref:
+                        text = match.get('matched_text')
```

Comment:
I think we would want the matched text in most cases, unless this is matched to a full license text for a standard SPDX license. For all other cases (notices, license_ref, etc.) we should get this.

Reply:
Correct me if I'm wrong, but isn't this what happens here? If we have a LicenseRef-, i.e. not an SPDX license, we enter the if clause and get the matched text.
This behavior is the same as in the previous implementation, so if you think this is a problem, I'd like to open a separate issue for this as this is not really part of this PR.

Reply:
Ack. Thanks for the clarification